### ISSUE TITLE
Cannot open include file: 'fficonfig.h' while building with msvc and ms-mpi

### ISSUE THREAD

**Issue by bzindovic** [rsmpi/rsmpi#163]
I cannot seem to build the library using the msvc toolchain. I have installed, rust-msvc-toolchain, msvc build tools with clang (all are from visual studio installer), ms-mpi and have set all paths but it always fails with the following message:
`libffi/src/x86/win64_intel.S(2): fatal error C1083: Cannot open include file: 'fficonfig.h': No such file or directory`

I was trying to build the example from `Readme.md` with following options in `Cargo.toml`:

`[dependencies]`
`mpi = { version = "0.6", features = ["user-operations", "derive"] }`

I've even tried turning off default features with `mpi = { version = "0.6.0", default-features = false }` and playing with the settings for the `libffi` crate but the same error appears.

Is there anyone who successfully built the example under Windows 10 x64 with ms-mpi and msvc toolchain?

**Comment by jedbrown** [comment]
Can you compare what you're doing to the Windows/MS-MPI job that runs in CI?
https://github.com/rsmpi/rsmpi/blob/main/.github/workflows/test.yaml#L58-L76

**Comment by bzindovic** [comment]
Yes, I've got everything set in the Visual Studio x64 Native development prompt for VS2022 (all ms-mpi variables are set, all clang dlls are visible in the prompt).

One thing to note, I'm using rust v1.69.0 which is built using llvm 15 (visual studio installs this version). You seem to be using an older version llvm  -14.0.6?

**Comment by bzindovic** [comment]
I've tried installing llvm 15.0.7 and tried using the latest git commit gave same results as before about the missing `fficonfig.h`.

**Comment by jedbrown** [comment]
I don't personally use Windows and don't have experience debugging this particular issue via email. I wonder if anyone else in the community can reproduce or if you can replicate in an environment that's possible to use in CI. (I don't really understand how it's 2023 and Windows is still so lousy at virtualizing.) Sorry I can't be of more help at the moment.

**Comment by bzindovic** [comment]
Thank you @jedbrown for giving me a hand, it seems that Windows environment has to have some strange workarounds (and this seems to be like that). I'll try investigating this more and will write here as soon as I find a workable and reproducible solution.

**Comment by bzindovic** [comment]
Just an update on the Windows compilation. Today, I had a successful build of the `rsmpi` under windows. I'll outline the steps I've used to build (I did it on several computers and it worked on all of them):
1. install visual studio build tools (or visual studio with c++ workload)
2. install ms-mpi (sdk and runtime)
3. install clang 15.0.7 from the official LLVM webpage (visual studio version didn't work for me)
4. reboot
6. open visual studio developer prompt and set the following variables:
`MSMPI_BIN=C:\Program Files\Microsoft MPI\Bin\`
`MSMPI_INC=C:\Program Files (x86)\Microsoft SDKs\MPI\Include\`
`MSMPI_LIB64=C:\Program Files (x86)\Microsoft SDKs\MPI\Lib\x64\`
7. Do the `cargo build` on your project

**Comment by bzindovic** [comment]
Another (rather strange) update. After updating rust to 1.70 on one of my computers, the original problem resurfaced. After trying various LLVM versions, I reverted to 1.69 and LLVM 15.0.7 but the problem persists. Really strange as the other computers (ones that I didn't update) work. 🤔😔😕

**Comment by jedbrown** [comment]
I've seen intermittent failures for parallel tests on Windows CI, which produces errors like this.
```
 Running: "mpiexec" "-n" "4" "D:\\a\\rsmpi\\rsmpi\\target\\debug\\examples\\structured"

job aborted:
[ranks] message

[0-1] process exited without calling init

[2-3] process exited without calling finalize

---- error analysis -----

[0-1] on fv-az251-990
D:\a\rsmpi\rsmpi\target\debug\examples\structured ended before calling init and may have crashed. exit code -1

[2-3] on fv-az251-990
D:\a\rsmpi\rsmpi\target\debug\examples\structured ended prematurely and may have crashed. exit code -1

---- error analysis -----
ERROR: Failed to post close command error 1726
ERROR: unable to tear down the job tree. exiting...
```
Re-running the jobs will sometimes succeed and sometimes fail on a different example. It seems somewhat episodic, as though the failure probability depends on other aspects of the GitHub actions infrastructure. Do you have ideas about this or should I assume it's a peculiarity of Actions (and/or MS-MPI bug) and just re-run until success?

**Comment by jedbrown** [comment]
I think this issue is resolved in 0.7.0. Feel free to reopen if not.
